Debugger/src/Connection/Firebase.php

```diff
@@ -0,0 +1,202 @@
+<?php
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Debugger\Connection;
+
+use Google\Cloud\Core\ArrayTrait;
+use Kreait\Firebase\Factory;
+use Kreait\Firebase\Database;
+
+/**
+ * Implementation of the Firebase connection
+ */
+class Firebase implements ConnectionInterface
+{
+    use ArrayTrait;
+
+    /**
+     * @var Factory
+     */
+    private $factory;
+
+    /**
+     * @var Database
+     */
+    private $database;
+
+    /**
+     * @var array
+     */
+    private $breakpoints;
+
+    public function __construct(array $config = [])
+    {
+        $databaseUrl = isset($config['firebase_db_url'])
+            ? $config['firebase_db_url']
+            : null;
+
+        if ($databaseUrl === null) {
+            $databaseUrl = 'https://' . $config['projectId'] . '-cdbg.firebaseio.com';
+        }
+
+        $this->factory = (new Factory())
```

Comment:
Would like some explanation for this as well(given that it's an external unofficial dependency).

Reply:
@carlos-granados can you explain why we need the external dependency? This seems like maybe something we wouldn’t want to include. Especially since we distribute our own firebase client library.

Reply:
When I searched for a library to interact with Firebase in PHP I was not able to find any "official" Google library and this is why I ended up using this external dependency. If there is such thing as an official Google Firebase client library in PHP can you please point me towards it?

Reply:
I assume we can use Firestore instead of firebase? https://github.com/googleapis/google-cloud-php-Firestone

Reply:
Unfortunately no, the open source tool https://github.com/GoogleCloudPlatform/snapshot-debugger uses the Firebase Realtime Database, not Firestore

Reply:
Understood, then the solution you have here may be our best option!